contracts/NFTAuction.sol

```diff
@@ -163,6 +169,38 @@ contract NFTAuction {
       ║          MODIFIERS          ║
       ╚═════════════════════════════╝*/
 
+    modifier contractHoldsNft(address _nftContractAddress, uint256 _tokenId) {
+        require(
+            IERC721(_nftContractAddress).ownerOf(_tokenId) == address(this),
+            "NFT not in contract"
+        );
+        _;
+    }
+
+    modifier isAuctionNotStartedByOwner(
```

Comment:
Here we check that the nft wasn't already started by the owner, and if there's a different owner we reset the auction params